libs/x-content/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -253,6 +255,10 @@ public XContentBuilder startObject() throws IOException {
         return this;
     }
 
+    public XContentBuilder startObject(ParseField field) throws IOException {
```

Comment:
I was thinking of adding these helpers, I'm glad you did :)

Reply:
++ have to check we have consistent for other methods - may be  add some more - just added most used like [Bb]oolean, [Ii]nt, [Ll]ong, String and objects